examples/manual-p2p.rs

```diff
@@ -0,0 +1,152 @@
+//! Low level example of connecting to and communicating with a peer.
+//!
+//! Run with
+//!
+//! ```not_rust
+//! cargo run --example manual-p2p
+//! ```
+
+use std::time::Duration;
+
+use colored::*;
+use futures::StreamExt;
+use reth_discv4::{Discv4, Discv4ConfigBuilder, DEFAULT_DISCOVERY_ADDRESS, DEFAULT_DISCOVERY_PORT};
+use reth_ecies::{stream::ECIESStream, util::pk2id};
+use reth_eth_wire::{
+    EthMessage, EthStream, HelloMessage, P2PStream, Status, UnauthedEthStream, UnauthedP2PStream,
+};
+use reth_network::config::rng_secret_key;
+use reth_primitives::{mainnet_nodes, Chain, Hardfork, Head, NodeRecord, MAINNET, MAINNET_GENESIS};
+use secp256k1::{SecretKey, SECP256K1};
+use tokio::net::TcpStream;
+
+type AuthedP2PStream = P2PStream<ECIESStream<TcpStream>>;
+type AuthedEthStream = EthStream<P2PStream<ECIESStream<TcpStream>>>;
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    // Setup discovery v4 protocol to find peers to talk to
+    let mut discv4_cfg = Discv4ConfigBuilder::default();
+    discv4_cfg.add_boot_nodes(mainnet_nodes());
+    discv4_cfg.lookup_interval(Duration::from_secs(1));
+
+    // Setup configs related to this 'node'
+    let our_key = rng_secret_key();
+    let our_peer_id = pk2id(&our_key.public_key(SECP256K1));
```

Comment:
nit we should add a `let (sk, pk) = keygen()`-style function to make this easier

Reply:
`let (secret_key, pk) = SECP256K1.generate_keypair(&mut rng);` exists, so can probably just wrap that

Reply:
(pk2id is different though)